src/core/src/package_managers/AptitudePackageManager.py

```diff
@@ -524,3 +539,36 @@ def update_os_patch_configuration_sub_setting(self, patch_configuration_sub_sett
     def do_processes_require_restart(self):
         """Defaulting this for Apt"""
         return False
+
+    def is_reboot_pending(self):
+        """Ubuntu Pro Client implementation for reboot pending"""
+        # If pre-requisite is False, raise exception to fall back
+        # to existing way in package_manager
+        if not self.__pro_client_prereq_met:
+            self.composite_logger.log_debug("Ubuntu Pro Client pre-requisite not met.")
+            return False, False
+        return self.ubuntu_pro_client.is_reboot_pending()
+
+    def __is_pro_client_prereq_met(self):
+        """check required conditions to use Ubuntu Pro Client"""
+        if Constants.UbuntuProClientSettings.FEATURE_ENABLED \
+                and self.__get_os_major_version() <= Constants.UbuntuProClientSettings.MAX_OS_MAJOR_VERSION_SUPPORTED \
+                and self.ubuntu_pro_client.is_pro_working()\
+                and self.__is_minimum_required_python_installed():
+            return True
+        else:
+            # Log each condition for debug.
+            self.composite_logger.log_debug("Feature_Enabled = {0}, OS Version = {1}, python {2}".format(Constants.UbuntuProClientSettings.FEATURE_ENABLED, Constants.UbuntuProClientSettings.FEATURE_ENABLED, self.__get_os_major_version() <= Constants.UbuntuProClientSettings.MAX_OS_MAJOR_VERSION_SUPPORTED, self.__is_minimum_required_python_installed()))
+            return False
+
+    def __get_os_major_version(self):
+        """get the OS major version"""
+        # Sample output for linux_distribution():
+        # ['Ubuntu', '20.04', 'focal']
+        os_version = self.env_layer.platform.linux_distribution()[1]
+        os_major_version = int(os_version.split('.')[0])
+        return os_major_version
+
+    def __is_minimum_required_python_installed(self):
+        """check if python version is at least 3.5"""
+        return sys.version_info >= (3, 5)
```

Comment:
belongs in that enum in Constants (the min version)

Reply:
Minor: Original comment author needs to see the change and then resolve the conversation. Otherwise it's difficult to track if a change was actually made and in the way that was discussed or expected. Unresolving for now as I don't see the change in the latest iteration yet.

Reply:
Got. it. Thanks.